### DISCUSSION TITLE
can we zoom the player's view ?

### DISCUSSION THREAD

**Discussion by BuzzLEquerre** [flareteam/flare-game#966] [General]
In dungeons or just in the map the player sees a lot of things and I would like for my game the camera to be more zoomed in on the player to have a reduced field of vision like that :
![Uploading Capture d'écran 2024-08-21 191020.png…]()

**Comment by dorkster** [comment]
You can try changing the `virtual_height` property in `engine/resolutions.txt`. This affects the both the game world and the UI. For fantasycore assets, the value shouldn't be any lower than 480 in order to properly fit the menus. You may need to resize your world map book to fit the lower resolution.

**Reply by BuzzLEquerre** [reply]
I managed to replace the dialogues with the npcs, but for my map I therefore replaced everything with -30% in my coordinates and I don't know why but there is a shift of the buttons on the 3rd line of my map, If you have any idea, knowing that I also reduced the images of the map and the button by 30% :

[NE.txt](https://github.com/user-attachments/files/16714132/NE.txt)
[NW.txt](https://github.com/user-attachments/files/16714134/NW.txt)
[SE.txt](https://github.com/user-attachments/files/16714135/SE.txt)
[SW.txt](https://github.com/user-attachments/files/16714136/SW.txt)
![Capture d'écran 2024-08-22 171733](https://github.com/user-attachments/assets/d5cafae2-f1ff-4db0-b1f2-e395e8a8f707)
![Capture d'écran 2024-08-22 171753](https://github.com/user-attachments/assets/5de2cc68-d03a-4762-be84-2e857d8b313a)

**Reply by dorkster** [reply]
First, I would make sure that the width and height in the `pos` property of each map matches the dimensions of each map image. Otherwise, the map might be stretched a little.

Looking at NE.txt, I see that the X value for the `button_pos` properties differs between rows . For example, row 1 column 1 is X=17, but row 3 column 1 is X=24. Make them all the same and they should line up.

**Reply by BuzzLEquerre** [reply]
That's good, thank you, and I asked you if it is possible to display an image as well as a description for each button without erasing the map in the background. Like that :
![Capture d'écran 2024-08-17 125015](https://github.com/user-attachments/assets/a52ae59f-fcc9-4a32-a9c9-8a948899234e)
(the game is Exiled Kingdom)

**Reply by dorkster** [reply]
Does that little info box show when you **hover** over each map tile? If so, you can't exactly replicate that, since we only let modders do stuff when the button is **clicked**.

You could work around this by requiring two clicks to travel: one to select the tile and one on a "Travel" button in the info box.

- First, the player would click on a map tile. This would do a `set_status` to indicate that the player selected that particular tile. You'd probably also want to do a `set_status` to indicate that *any* tile was selected.
- You can then use those statuses (with `requires_status` to control the visibility of text, image, and button elements to form the info box. The info box background and the button(s) should be controlled with the "any tile" status, while the preview image and text can be controlled by the individual tile statuses.
- It's handy to have a script to `unset_status` all the statuses that you use when you open/close the book. We do this for the alchemy/crafting menu in the Empyrean campaign. You might want to dig into that to see how I approached this sort of interactivity.

**Reply by BuzzLEquerre** [reply]
I have a few questions,
the first, do events work with buttons that set a set_status?

Then I tried to do what you said with a double click system, but I don't understand how to make the second click work by showing the tile info box.

And I would like to know if the visibility commands you talked about are "image_visibility=100%", "image_button_visibility=100%" and "text_visibility=100%"

This is what I tried to do, with an event and a button :
[NE.txt](https://github.com/user-attachments/files/16732146/NE.txt)

**Reply by dorkster** [reply]
Events can read any status set by a button with `set_status`. But you can't use `[event]` in a book. If you need to use multiple events for an action, you should use a script. In this case, you'll probably want a single script to handle traveling. Something like this:

```
# scripts/travel.txt
[event]
requires_status=0_0_NE
intermap=...

[event]
requires_status=0_1_NE
intermap=...
```

Back to the map, you can control the visibility of `[image]` and `[text]` elements with `requires_status`. If the status check fails, the image/text won't be shown. You can also disable `[button]` elements with the same method. Buttons will always be visible, but they won't be able to be clicked. Here's what this might look like:

```
# these elements are shared for all tile previews
[image]
image=<filename>
image_pos=<X>,<Y>
requires_status=travel_preview

[button]
button_image=<filename>
button_text=Travel
# this button position should be in the info box, like where "Back to game" is in your example screenshot
button_pos=<X>,<Y>
requires_status=travel_preview
script=scripts/travel.txt

# these elements depend on which tile is selected
# thumbnail preview image of region
[image]
image=<filename>
image_pos=<X>,<Y>
requires_status=0_0_NE

[text]
text=...
text_pos=<X>,<Y>,<Width>,<Justify>

### Begin tiles
# 0/0
[button]
button_pos=17,17
button_image=images/menus/buttons/map.png
set_status=0_0_NE,travel_preview

**Reply by BuzzLEquerre** [reply]
Hi, Coming back to my question, I would like to know if changing the minimum value of "virtual_height" by replacing 480 -> 300 or 120 would involve recoding a lot of elements of the game engine or if it is "simple".

**Reply by dorkster** [reply]
The main thing you'd need to deal with is the UI. If you're using the fantasycore UI either verbatim or as a template, it assumes a minimum vertical resolution of 480 pixels. So if you want to go smaller, you'll need to change the images and layouts for most of the menus.
